armory/data/utils.py

```diff
@@ -81,6 +84,29 @@ def maybe_download_weights_from_s3(weights_file: str) -> str:
                         "`saved_model_dir` directory on your host environment."
                     )
                 )
+
+    if auto_expand_tars:
+        if tarfile.is_tarfile(filepath):
+            logger.debug(f"Detected model weights file {weights_file} as a tar archive")
+            with tarfile.open(filepath) as tar:
+                # check if the tarfile contains a directory containing all its members
+                # ie if the tarfile expands out entirely into a subdirectory
+                dirs = [fi.name for fi in tar.getmembers() if fi.isdir()]
+                for d in dirs:
+                    if all(name.startswith(d) for name in tar.getnames()):
+                        subdir = d
+                        break
+                else:
+                    raise PermissionError(
+                        (
+                            f"{weights_file} does not expand into a subdirectory."
+                            f" Weights files submitted as tarballs must expand into a subdirectory."
+                        )
+                    )
+                logger.info(f"Auto-unpacking model weights from {weights_file}")
+                tar.extractall(path=saved_model_dir)
```

Comment:
Some of these tar files are bound to be rather large - it would be rather painful, I think, to have to expand them each time you run armory. If running locally, you will repeatedly extract/overwrite those files.

I would rather check to see if the subdir exists and then log a warning like `f"Subdir already exists. Not expanding tar. Must remove {os.path.join(saved_model_dir, subdir)} manually."` and then skip extraction.